charmhelpers/contrib/hahelpers/apache.py

```diff
@@ -64,6 +70,36 @@ def get_cert(cn=None):
     return (cert, key)
 
 
+def validate_cert(cert, key, ca):
+    """
+    cert (bytes): cert.crt file contents (PEM armored)
+    key (bytes): cert.key file contents
+    ca (Optional[bytes]): None or private ca contents
+
+    return True if ok
+    otherwise return False, set blocked status, and set useful error message
+    """
+
+    context = None
+    if ca:
+        context = ValidationContext(trust_roots=[ca])
+
+    # split certs so we can pass leaf and intermediate certs separately
+    # The leaf/server certificate _must_ be the first cert https://www.rfc-editor.org/rfc/rfc5246#page-48
+    certs = re.findall(r'-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----', cert.decode(), re.DOTALL)
```

Comment:
Yes, we can assume PEM format.
https://httpd.apache.org/docs/2.4/mod/mod_ssl.html#sslcertificatefile

Also, I found in the parent library that `multiple=True` can split the concatenated certs into an array in DER format.
https://github.com/wbond/asn1crypto/blob/master/docs/pem.md#unarmor
So we could use `certs[0]` as a leaf cert, then the rest `cert[1], [2], ...` can be used as an intermediate CA chain. We might be able to use DER as is since CertificateValidator states it can accept DER and PEM or we need to armor those back to PEM.

Reply:
Well, the second example in `Basic Path Validation` section is exactly for it :-)
https://github.com/wbond/certvalidator/blob/master/docs/usage.md#basic-path-validation